### PR TITLE
Fix STIX2 1 object.

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19976 https://github.com/demisto/etc/issues/19647

## Description
A few sentences describing the overall goals of the pull request's commits.

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Scripts/StixParser/StixParser.py [hunk 1]
```diff
@@ -274,12 +274,30 @@ def extract_indicators(data):
...
+    if isinstance(data, dict):
+        # Check if is `objects` keyword exists
+        if "objects" in data:
+            objects = data.get("objects")
+        # If its STIX
+        elif all([key in data for key in must_have_in_stix]):
+            objects = data
+        else:
+            return_error("No STIX2 object could be parsed")
     if isinstance(data, dict) and "objects" in data:
```
Comment (Itay4) [inline Scripts/StixParser/StixParser.py]: why are you checking this again ?

**Comment by jochman** [comment]
image
server passed tests

**Comment by Itay4** [comment]
@jochman please add test

**Comment by Itay4** [comment]
@jochman did @kirbles19 review this ?

File: Scripts/StixParser/StixParser_test.py [hunk 2]
```diff
@@ -219,3 +219,10 @@ def test_main(self, mocker):
         files_path = "./TestData/stix1/"
         mock_demisto(mocker)
         self._run_on_files(files_path, main, mocker)
+
+
+def test_dict_no_stix(mocker):
+    from StixParser import stix2_to_demisto
+    mock_demisto(mocker)
+    with pytest.raises(SystemExit):
```
Comment (Itay4) [inline Scripts/StixParser/StixParser_test.py]: check if `return_error` is called instead, and verify it was called with `"No STIX2 object could be parsed"`

**Comment by jochman** [comment]
> @jochman did @kirbles19 review this ?

yes, on the previous pr

**Pull request merged**